azuredocumentdb/src/main/java/com/yahoo/ycsb/db/azuredocumentdb/AzureDocumentDBClient.java

```diff
@@ -0,0 +1,230 @@
+  }
+
+  @Override
+  public Status update(String table, String key, HashMap<String, ByteIterator> values) {
+    Document record = getDocumentById(table, key);
+
+    if (record == null) {
+      return Status.ERROR;
+    }
+
+    // Update each field.
+    for (Entry<String, ByteIterator> val : values.entrySet()) {
+      record.set(val.getKey(), val.getValue().toString());
+    }
+
+    // Replace the document.
+    try {
+      documentClient.replaceDocument(record, null);
+    } catch (DocumentClientException e) {
+      e.printStackTrace();
+      return Status.ERROR;
+    }
+
+    return Status.OK;
+  }
+
+  @Override
+  public Status insert(String table, String key, HashMap<String, ByteIterator> values) {
+    Document record = new Document();
+
+    record.set("id", key);
+
+    for (Entry<String, ByteIterator> val : values.entrySet()) {
+      record.set(val.getKey(), val.getValue().toString());
+    }
+
+    try {
+      documentClient.createDocument(collection.getSelfLink(), record, null, false);
+    } catch (DocumentClientException e) {
+      e.printStackTrace();
+      return Status.ERROR;
+    }
+    return Status.OK;
+  }
+
+  @Override
+  public Status delete(String table, String key) {
+    Document record = getDocumentById(table, key);
+
+    try {
+      // Delete the document by self link.
+      documentClient.deleteDocument(record.getSelfLink(), null);
+    } catch (DocumentClientException e) {
+      e.printStackTrace();
+      return Status.ERROR;
+    }
+
+    return Status.OK;
+  }
+
+  @Override
+  public Status scan(String table, String startkey, int recordcount,
+                     Set<String> fields, Vector<HashMap<String, ByteIterator>> result) {
+    // TODO: Implement Scan as query on primary key.
+    return Status.NOT_IMPLEMENTED;
+  }
+
+  private Database getDatabase() {
+    if (database == null) {
+        // Get the database if it exists
+        List<Database> databaseList = documentClient
+                .queryDatabases(
+                        "SELECT * FROM root r WHERE r.id='" + databaseId
+                                + "'", null).getQueryIterable().toList();
+
+        if (databaseList.size() > 0) {
+            // Cache the database object so we won't have to query for it
+            // later to retrieve the selfLink.
+            database = databaseList.get(0);
+        } else {
+            // Create the database if it doesn't exist.
+            try {
+                Database databaseDefinition = new Database();
+                databaseDefinition.setId(databaseId);
+
+                database = documentClient.createDatabase(
+                        databaseDefinition, null).getResource();
+            } catch (DocumentClientException e) {
+                // TODO: Something has gone terribly wrong - the app wasn't
+                // able to query or create the collection.
+                // Verify your connection, endpoint, and key.
+                e.printStackTrace();
```

Comment:
These errors should be sent to stderr. An example is here:
```
e.printStackTrace(System.err);
```

Another option would be to use a logging framework but that would be a bigger change.